src/blob/core.rs

```diff
@@ -251,8 +250,8 @@ where
         debug!("blob write");
         // Only one upgradable_read lock is allowed at a time
         let (partially_serialized, header) = record.to_partially_serialized_and_header()?;
-        let blob = blob.upgradable_read().await;
-        Self::write_locked(blob, key, partially_serialized, header).await
+        let blob = blob.read().await;
+        Self::write_with_lock(blob, key, partially_serialized, header).await
```

Comment:
I think the code from `write_with_lock` can be moved in here and `write_with_lock` can be removed